R/utils.R

```diff
@@ -60,53 +60,43 @@ get_single_id <- function(param_list) {
     paste(sep = ".", collapse = ".")
 }
 
-#' Make a forecaster grid.
-#'
-#' Convert a tibble of forecasters and their parameters to a specific format
-#' that we can iterate over in targets. Currently only `forecaster` and
-#' `trainer` can be symbols.
-#'
-#' @param tib the tibble of parameters. Must have the forecaster and trainer
-#' columns, everything else is optional.
-#'
-#' @export
-make_forecaster_grid <- function(tib) {
-  if ("trainer" %in% colnames(tib)) {
-    tib$trainer <- rlang::syms(tib$trainer)
-  }
-  # turns a tibble into a list of named lists
-  params_list <-
-    tib %>%
+#' Turn a tibble of parameters into a list of named lists.
+make_params_list <- function(df, unlist_cols = c("lags", "trainer"), get_cols = c("trainer")) {
+  params_list <- df %>%
     select(-forecaster, -id) %>%
     split(seq_len(nrow(.))) %>%
     unname() %>%
     lapply(as.list)
-  # for whatever reason, trainer ends up being a list of lists, which we do not want
-  params_list %<>% lapply(function(x) {
-    x$trainer <- x$trainer[[1]]
-    x$lags <- x$lags[[1]]
-    x
-  })
-
-  if (length(params_list) == 0) {
-    out <- tibble(
-      id = tib$id,
-      forecaster = rlang::syms(tib$forecaster),
-      params = list(list()),
-      param_names = list(list())
-    )
-  } else {
-    out <- tibble(
-      id = tib$id,
-      forecaster = rlang::syms(tib$forecaster),
-      params = params_list,
-      param_names = map(params_list, names)
-    )
+  names(params_list) <- df$id
+
+  # Some columns need to be unlisted.
+  unlist_cols <- unlist_cols[unlist_cols %in% names(params_list[[1]])]
+  if (length(unlist_cols) > 0) {
+    params_list %<>% lapply(function(x) {
+      for (col in unlist_cols) {
+        if (length(x[[col]]) == 1) {
+          x[[col]] <- x[[col]][[1]]
+        }
+      }
+      x
+    })
+  }
+
+  # Some columns need to be converted to symbols.
+  get_cols <- get_cols[get_cols %in% names(params_list[[1]])]
+  if (length(get_cols) > 0) {
+    params_list %<>% lapply(function(x) {
+      for (col in get_cols) {
+        x[[col]] <- get(x[[col]])
+      }
+      x
+    })
```

Comment:
This function changed, since the way we fed arguments to functions changed. Mostly, we didn't need to `symbol` things anymore, since we targets won't ruin things, so `trainer` can just be a function/object.

Reply:
reverted